hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java

```diff
@@ -320,6 +326,57 @@ public void testAlreadyExistsException() {
         () -> catalog.createTable(TABLE_IDENTIFIER, schema, PartitionSpec.unpartitioned()));
   }
 
+  @Test
+  public void testCreateTableCommitThrowsUnknownException()
+      throws TException, InterruptedException {
+    String namespace = DB_NAME;
+    String tableName = getRandomName();
+    TableIdentifier tableIdentifier = TableIdentifier.of(namespace, tableName);
+
+    HiveTableOperations hiveTableOperations =
+        (HiveTableOperations) catalog.newTableOps(tableIdentifier);
+
+    String tableLocation = catalog.defaultWarehouseLocation(tableIdentifier);
+    TableMetadata metadataV1 = createTableMetadata(tableLocation);
+
+    HiveTableOperations spyOps = spy(hiveTableOperations);
+    failCommitAndThrowException(spyOps);
+
+    Assertions.assertThatThrownBy(() -> spyOps.commit(null, metadataV1))
+        .isInstanceOf(RuntimeException.class)
+        .hasMessageContaining("Datacenter on fire");
+
+    Assert.assertEquals(
+        "One metadata files should exist",
+        1,
+        metadataFileCount(tableLocation + "/writeMetaDataLoc"));
```

Comment:
Could we just test the default Hive locations instead of changing the default behaviour and  a specifying the locations by table properties?

Reply:
Why can not we use the `metadataV1` and `metadataFileCount(metadataV1)` to get the file number?

Reply:
> Why can not we use the metadataV1 and metadataFileCount(metadataV1)

Since the purpose of the test is to create a table, I am creating a new `TableMetadata` object using the `createTableMetadata` method. However, this method does not set the `metadataFileLocation` member variable of the `TableMetadata` object. As a result, I cannot use the `metadataFileCount(metadataV1)` method because it calls `metadata.metadataFileLocation()`, which returns null.

> Could we just test the default Hive locations

I have to pass the metadata location to get the file count. metadata location is specified in `createTableMetadata`.

I will try to see if we can use the existing `metadataFileCount(metadataV1)` method.

Reply:
Also, note that `TableMetadata#metadataFileLocation` will not be set with the new metadata file location until the table is created. In this PR case, the table will not be created as it fails during `commit`.

Reply:
Sorry for the late answer!
Could we use assumptions about the table location instead in the tests? I am afraid that the default behavior is different from the behavior when the location is set, and with this tests we test only the second scenario. OTOH creating a Table without specifically setting the location is the way we use Iceberg in Hive. So it would be good to test that case as well.

Thanks again for taking care of this!